rotary/rules/polish.smk

```diff
@@ -15,37 +15,120 @@ rule prepare_medaka_polish_input:
     input:
         "{sample}/assembly/{sample}_assembly.fasta"
     output:
-        temp("{sample}/polish/medaka_input/{sample}_input.fasta")
+        fasta=temp("{sample}/polish/medaka_input/{sample}_input.fasta"),
+        fasta_dir=temp(directory("{sample}/polish/medaka_input/"))
     run:
-        source_relpath = os.path.relpath(str(input),os.path.dirname(str(output)))
-        os.symlink(source_relpath,str(output))
-
+        source_relpath = os.path.relpath(str(input),os.path.dirname(str(output.fasta)))
+        os.symlink(source_relpath,str(output.fasta))
 
-rule polish_medaka:
+rule map_long_reads_for_medaka:
     input:
         qc_long_reads="{sample}/qc/{sample}_qc_long.fastq.gz",
         contigs="{sample}/{step}/medaka_input/{sample}_input.fasta"
     output:
-        dir=directory("{sample}/{step}/medaka"),
-        contigs="{sample}/{step}/medaka/{sample}_consensus.fasta",
-        calls_to_draft=temp(multiext("{sample}/{step}/medaka/calls_to_draft", '.bam', '.bam.bai')),
-        consensus_probs=temp("{sample}/{step}/medaka/consensus_probs.hdf")
+        calls_to_draft_bam=temp('{sample}/{step}/medaka/calls_to_draft.bam'),
+        calls_to_draft_bam_index=temp('{sample}/{step}/medaka/calls_to_draft.bam.bai')
     conda:
         "../envs/medaka.yaml"
     log:
-        "{sample}/logs/{step}/medaka.log"
+        "{sample}/logs/{step}/map_long_reads_medaka.log"
+    benchmark:
+        "{sample}/benchmarks/{step}/map_long_reads_medaka.txt"
+    threads:
+        config.get("threads",1)
+    params:
+        output_base_name='{sample}/{step}/medaka/calls_to_draft'
+    shell:
+        """
+        mini_align -i {input.qc_long_reads} -r {input.contigs} -m -p {params.output_base_name} -t {threads} > {log} 2>&1
+        """
+
+checkpoint generate_contig_manifest:
+    input:
+        "{sample}/{step}/medaka_input/{sample}_input.fasta"
+    output:
+        contig_manifest="{sample}/{step}/medaka/{sample}_contigs.txt"
+    conda:
+        "../envs/medaka.yaml"
+    log:
+        "{sample}/logs/{step}/contig_names_medaka.log"
     benchmark:
         "{sample}/benchmarks/{step}/medaka.txt"
+    shell:
+        """
+        seqkit seq -n {input} > {output.contig_manifest} 2>> {log}
+        """
+
+rule polish_contig_medaka:
+    input:
+        calls_to_draft_bam='{sample}/{step}/medaka/calls_to_draft.bam',
+        calls_to_draft_bam_index='{sample}/{step}/medaka/calls_to_draft.bam.bai',
+        contig_manifest="{sample}/{step}/medaka/{sample}_contigs.txt"
```

Comment:
It seems like this input (`contig_manifest`) is not called in the shell section below -- is that correct?

Reply:
Addressed